source/frontend/src/features/leases/detail/LeasePages/payment/modal/period/PeriodForm.test.tsx

```diff
@@ -83,6 +84,32 @@ describe('PeriodForm component', () => {
     expect(getByText('Add Variable Rent')).toBeVisible();
   });
 
+  it('displays variable period gst when manual gst does not equal calculated gst.', async () => {
+    const { getByDisplayValue, getByTestId } = setup({
+      initialValues: {
+        ...defaultFormLeasePeriod,
+        isGstEligible: true,
+        paymentAmount: 1,
+        variableRentPaymentAmount: 1,
+        additionalRentPaymentAmount: 1,
+        isVariableRentGstEligible: true,
+        isAdditionalRentGstEligible: true,
+        variableRentGstAmount: 100,
+        gstAmount: 200,
+        additionalRentGstAmount: 300,
+      },
+    });
+
+    const variableButton = getByTestId('radio-isvariable-variable');
+    await act(async () => {
+      userEvent.click(variableButton);
+    });
+
+    expect(getByDisplayValue('$100.00')).toBeVisible();
+    expect(getByDisplayValue('$200.00')).toBeVisible();
+    expect(getByDisplayValue('$300.00')).toBeVisible();
```

Comment:
Not quite fond of this pattern since it could lead to false positives when any other element displays the same value. Preferably find the element in cuestion and use toHaveValue to assert results explicitly.